### ISSUE TITLE
Request switcher keyboard shortcuts not working

### ISSUE THREAD

**Issue by JeroenVdb** [getinsomnia/insomnia#6301]
### Expected Behavior

My keyboard shortcuts/keybinding are not working anymore. For example hitting Command + P does not open the Switch Requests	pallet. Command + , does not open the Insomnia settings.

### Actual Behavior

Nothing happens.

### Reproduction Steps

1. Press Command + ,
2. Nothing happens

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

This worked before. AFAIK it stopped working since I updated to v2023.5.2 this morning.

### Insomnia Version

2023.5.2

### What operating system are you using?

macOS

### Operating System Version

Venture 13.5 (22G74)

### Installation method

Downloaded from Insomnia.rest website

### Last Known Working Insomnia version

< 2023.5.2 (but don't know specific version, I'm a regular updater so I guess the previous version)

**Comment by jackkav** [comment]
The request switcher was spaghetti code so we dropped it to unblock our refactor, it will be back soon. The settings hotkey works on my machine.

**Comment by JeroenVdb** [comment]
> The request switcher was spaghetti code so we dropped it to unblock our refactor, it will be back soon.

Clear, hope it's back soon. I mainly used it to quickly switch between collections. It was the closest to a general Command Pallet in the app.

> The settings hotkey works on my machine.

`Command + ,` still does not work on my machine. But `Command + Shift + ,` does work. I'm on an AZERTY keyboard. Maybe that explains it?

**Comment by jackkav** [comment]
We are thinking about introducing exactly that kind of command palette idea!

That would certainly explain it.
I'll keep this issue open until switcher is back

**Comment by Caskman** [comment]
This is not a refactor, this is a major version change since features were dropped.  Ctrl + Tab is extremely useful when I have a lot of requests and I'm switching between requests on opposite ends of the list.

I know I can switch to a previous version but this makes me much more reluctant to update going forward

**Comment by drahnieR** [comment]
> This is not a refactor, this is a major version change since features were dropped.

And I'm surprised that it's not mentioned in the changelog.

**Comment by johnli2210** [comment]
For anyone looking to rollback I tested each GA release and 2023.4.0 appears to be the last stable version with a working Request Switcher: https://github.com/Kong/insomnia/releases/tag/core%402023.4.0

I for one find this feature essential!

**Comment by nikbrunner** [comment]
> For anyone looking to rollback I tested each GA release and 2023.4.0 appears to be the last stable version with a working Request Switcher: https://github.com/Kong/insomnia/releases/tag/core%402023.4.0
> 
> I for one find this feature essential!

Thank you very much! This is also very essential for me!

Now <kbd>Cmd+p</kbd> works again, but what I find interesting is, that now <kbd>Cmd+Shift+e</kbd> (This opens the Env Switcher by default) is not working in this version (`2023.4.0`).

In the current version, it is the other way round.

**Comment by sebastianjonasson** [comment]
> Now Cmd+p works again, but what I find interesting is, that now Cmd+Shift+e (This opens the Env Switcher by default) is not working in this version (`2023.4.0`).

`Cmd+Shift+e` was patched in `2023.5.0`.

**Comment by nikbrunner** [comment]
> > Now Cmd+p works again, but what I find interesting is, that now Cmd+Shift+e (This opens the Env Switcher by default) is not working in this version (`2023.4.0`).
> 
> `Cmd+Shift+e` was patched in `2023.5.0`.

Yes, I know, but in `2023.5.0` is <kbd>Cmd+p</kbd> is not working. 😁

**Comment by VerinskiiNikita** [comment]
still not working 2023.5.7
cmd+p

**Comment by bukajsytlos** [comment]
this should be top priority...

**Comment by nikbrunner** [comment]
Version: `2023.5.8` 

These for me essential key binds are (still) not working:

- <kbd>cmd+p</kbd> - Browse Requests
- <kbd>cmd+shift+e</kbd> - Switch Environment

**Comment by xaviergmail** [comment]
For the longest time, I thought it had something to do with the vim keybinds option, or something wrong with my Karabiner configuration. I'm relieved to know it's not just me.

FWIW, neither cmd+p or cmd+shift+e are working on 8.0.0 (2023-09-27 build)

**Comment by relgames** [comment]
Hi, CMD+P is not working for me as well in 8.1.0:

Version: Insomnia 8.1.0
Build date: 9/28/2023

Seems like this issue is open for 1.5 months, can you just bring back the "spaghetti code" which worked and we liked it, and then *after* the refactoring, feel free to delete it?

**Comment by Lasse-BeCause** [comment]
Quickly tabbing between recent requests is super useful to me. I'm considering switching away from Insomnia because of this simple feature, that's how much I use tabbing between requests.

Btw the Preferences menu still has the keybind, so it's not clear to me whether this feature is supposed to work or be gone forever.

**Comment by almeidacavalcante** [comment]
I just updated and CMD+P ramains not working. I tried to change the shortcut but didn't have any luck. Do we know any insomnia version that works?

**Comment by Ivan-Donchenko** [comment]
@almeidacavalcante This one works for me https://github.com/Kong/insomnia/releases/tag/core%402023.4.0
It seems its last version with working request switcher

**Comment by VerinskiiNikita** [comment]
> The request switcher was spaghetti code so we dropped it to unblock our refactor, it will be back soon. The settings hotkey works on my machine.


So sorry that it is October now and shortcuts still not working. So sorry that I have to change platform to postman

**Comment by suchislife801** [comment]
## Last Working Version! Sort Order Fix!
### Also, everything that used to work, works on this version.
# [2023.4.0](https://github.com/Kong/insomnia/releases/tag/core@2023.4.0)
### ANY version ABOVE 2023.4.0 will absolutely WRECK your sort orders.

**Comment by mbcompass** [comment]
Please, bring back cmd+p and control+tab

**Comment by nedmules-sked** [comment]
Hi @jackkav and the Insomnia product managers

I have been unable to upgrade Insomnia past version 2023.4.0 because for me it is unusable without the Ctrl-Tab feature to switch between requests - I have 100's of requests in my workspace so scrolling or filtering is not an option. I want to upgrade so I can get bug fixes in later versions.

In August, Jack said:

> The request switcher was spaghetti code so we dropped it to unblock our refactor, it will be back soon. The settings hotkey works on my machine.

Can you tell us where the request switcher sits on your list of priorities and when you expect it to be fixed/restored? In my view, you should fix this major breaking change before adding any more features.

Thanks, Ned

**Comment by relgames** [comment]
Turned out, 2013.4.0 has another issue - unable to search environments, it just shows a big dropdown but not able to type to filter the dropdown list. 

Why do you keep breaking features? Just stop, it was working well and we were paying for it for 10 developers. Now it does not work and no one wants to use it anymore.

**Comment by subnetmarco** [comment]
As discussed in https://github.com/Kong/insomnia/discussions/6815#discussioncomment-7765772 let's see if we can get this fixed in our December release in a couple of weeks.

**Comment by subnetmarco** [comment]
This issue has been fixed and will ship in the next release of Insomnia in late January.

**Comment by subnetmarco** [comment]
Available in 8.5.2 beta, before GA (next week) we will also make it work across multiple organizations.

**Comment by JeroenVdb** [comment]
I'm using the beta, works great in general at first sight 💪

One remark: is it possible to already select the first item from the list? Now you have to press down to select/highlight the first item in the list. I watched some other command palettes (Chrome DevTools, Raycast, IntelliJ) and they all have the first item automatically selected.

**Comment by subnetmarco** [comment]
@JeroenVdb good feeback, yes we can do that.

**Comment by JeroenVdb** [comment]
Feedback about the filter being used, leaving it here because the feature is still in beta, but I can also create a new issue if that would be better.

I think [the current code for the filter](https://github.com/gatzjames/insomnia/blob/ab95f6c48d082f79fdb1c909f56ef4afa2edce6b/packages/insomnia/src/ui/components/command-palette.tsx#L48-L51) needs some improvements.

The current filter splits the filter query and adds wildcards between each character. Which leads to some weird items not being filtered out because they "accidentally" have those characters in that order somehow.

For example:
```
Search for "scen"

Returns:
- "GET /scenarios {{ _.ServiceBaseUrl }}/scenarios"
- "GET /changes {{ _.ServiceBaseUrl }}/changes"

I would expect the first one but not the second one.
```

If I remember correctly the previous search of the request switcher worked very accurate, meaning my searches most of the time narrowed the list down to the item I was looking for or maybe with at most 5 other possibilities showing.

The previous ["request-switcher" used the `fuzzyMatchAll` util function](https://github.com/Kong/insomnia/blob/d385ffa5d74f3251a32d697051bc0309e98c12af/packages/insomnia-app/app/ui/components/modals/request-switcher-modal.tsx#L220-L226C7). As [this function still exists](https://github.com/Kong/insomnia/blob/d385ffa5d74f3251a32d697051bc0309e98c12af/packages/insomnia-app/app/common/misc.ts#L284) it might be an easy change again use this way of filtering.

**Comment by bukajsytlos** [comment]
afair it worked like that before

**Comment by bukajsytlos** [comment]
does this also fix ctrl+tab?

**Comment by nikbrunner** [comment]
The default binding for switching Environments or a custom one is still not working. <kbd>cmd+p</kbd> works now though. 

<img width="756" alt="image" src="https://github.com/Kong/insomnia/assets/46924729/01698092-b658-4f0f-83aa-d6bd319f25aa">

**Comment by JeroenVdb** [comment]
I think it’s best to create a new issue for this. 

Also: I would rather add this functionality to the command palette instead of a separate modal. The shortcut could then open the command palette with a predefined filter. See https://github.com/Kong/insomnia/discussions/7030

**Comment by vytautasphp** [comment]
New version updated, but ctrl+tab still not working... unbelievable, Insomnia now has a lot of useless features, but super simple feature not working, wtf... Its so annoying...

**Comment by JeroenVdb** [comment]
@vytautasphp it might be a good idea to open a new issue, this one was very focused on switching between requests and collections. I think your issue is more about switching between the last X requests made.

**Comment by bukajsytlos** [comment]
sry, but this issue was mentioned here x times. if they did not want to implement it here, then they should made new issue...

**Comment by subnetmarco** [comment]
We re-introduced CMD + P. We will work on fixing CMD + SHIFT + E and CTRL+TAB, most likely it will ship in the next release.

I am tasking the team to go over the entire list of all shortcuts to check them all, as a matter of fact.

Apologies for the disruption, all these issues are getting improved on every release, we are doing our best to keep up.

**Comment by sevinchek** [comment]
any updates on this?

**Comment by subnetmarco** [comment]
@gatzjames any update here?
